kernel/task.h

```diff
@@ -188,4 +194,26 @@ extern void (*exit_hook)(struct task *task, int code);
 // Will ensure that the -pid part always fits, then will fit as much of comm as possible.
 void update_thread_name(void);
 
+// To collect statics on which tasks are blocked we need to proccess areas
+// of code which could block our task (e.g reads or writes). Before executing
+// of functions which can block the task, we mark our task as blocked and
+// unblock it after the function is executed.
+__attribute__((always_inline)) inline int enter_blocking_area(void) {
+    lock(&pids_lock);
+    current->io_block = 1;
+    blocked_task_count++;
```

Comment:
Instead of maintaining a counter which could get out of sync easily, I'd prefer to iterate over the process list and count the number of running processes when you read the proc file. There aren't going to be all that many of them. This goes for both `blocked_task_count` and `alive_task_count`.